### DISCUSSION TITLE
Move the API Provider Settings to JSON File Instead of Global State

### DISCUSSION THREAD

**Discussion by ghost** [cline/cline#380] [Feature Requests]
Right now, all the settings for the API provider are stored in the global state database (`getGlobalState` in `claude-dev/src/providers/ClaudeDevProvider.ts`). However, this setup becomes a bit cumbersome when the API endpoint changes. 

To update the settings, I currently have to manually clear the state from the database by running the following commands:

```
sqlite3 ~/Library/Application\ Support/Code/User/globalStorage/state.vscdb
SELECT * FROM ItemTable WHERE key LIKE "%saoudrizwan.claude-dev%";
```

Then I need to hard restart VS Code. This is also true for other settings like history, custom instructions, etc.

A more user-friendly solution would be to move these settings to a simple JSON file. This way, users can easily modify API settings without messing around with the global state. An alternative could be an options webview where we can directly update these settings through a UI, making things much more intuitive.

**Comment by xvweirong** [comment]
Just similar as the Json config file Continue use. It's quite straight forward.

**Comment by mentianyi** [comment]
It is a big feature.
